middlewares/authenticator.go

```diff
@@ -76,9 +88,20 @@ func parserBasicUsers(users types.Users) (map[string]string, error) {
 	return userMap, nil
 }
 
-func parserDigestUsers(users types.Users) (map[string]string, error) {
+func parserDigestUsers(digest *types.Digest) (map[string]string, error) {
+	var userStrs []string
+	if digest.UsersFile != "" {
+		var err error
+		if userStrs, err = getUserStringsFromFile(digest.UsersFile); err != nil {
+			return nil, err
+		}
+	}
+	userStrs = append(digest.Users, userStrs...)
 	userMap := make(map[string]string)
-	for _, user := range users {
+	for _, user := range userStrs {
+		if user == "" {
```

Comment:
Because the strings (lines from a file) that we get back from `getUserStringsFromFile()` can legitimately contain blank lines. Those need to be discarded since there's no username / password information we can parse out of an empty string.